Various debian-specific ExtUtils changes:

 * Respect umask during installation, and set as appropriate for each of
   perl, vendor and site (policy requires group writable site dirs).

 * Don't install .packlist or perllocal.pod for perl or vendor.
 * Fiddle with *PREFIX and variables written to the makefile so that
   install directories may be changed when make is run by passing
   PREFIX= to the "make install" command (used when packaging
   modules).

 * Set location of libperl.a to /usr/lib.
 * Note that libperl-dev package is required for embedded linking.

diff -Naur --exclude=debian perl-5.8.6.orig/lib/ExtUtils/Embed.pm perl-5.8.6/lib/ExtUtils/Embed.pm
--- perl-5.8.6.orig/lib/ExtUtils/Embed.pm	2002-02-10 06:09:16.000000000 +1100
+++ perl-5.8.6/lib/ExtUtils/Embed.pm	2005-05-21 14:59:33.000000000 +1000
@@ -297,6 +297,9 @@
 Typically, an application B<Makefile> will invoke ExtUtils::Embed
 functions while building your application.  
 
+Note that on Debian systems the B<libperl-dev> package is required for
+compiling applications which embed an interpreter.
+
 =head1 @EXPORT
 
 ExtUtils::Embed exports the following functions:
diff -Naur --exclude=debian perl-5.8.6.orig/lib/ExtUtils/Install.pm perl-5.8.6/lib/ExtUtils/Install.pm
--- perl-5.8.6.orig/lib/ExtUtils/Install.pm	2004-01-06 09:34:59.000000000 +1100
+++ perl-5.8.6/lib/ExtUtils/Install.pm	2005-05-21 15:00:42.000000000 +1000
@@ -173,8 +173,8 @@
 		if (-f $targetfile){
 		    forceunlink($targetfile) unless $nonono;
 		} else {
-		    mkpath($targetdir,0,0755) unless $nonono;
-		    print "mkpath($targetdir,0,0755)\n" if $verbose>1;
+		    mkpath($targetdir) unless $nonono;
+		    print "mkpath($targetdir)\n" if $verbose>1;
 		}
 		copy($sourcefile, $targetfile) unless $nonono;
 		print "Installing $targetfile\n";
diff -Naur --exclude=debian perl-5.8.6.orig/lib/ExtUtils/MM_Any.pm perl-5.8.6/lib/ExtUtils/MM_Any.pm
--- perl-5.8.6.orig/lib/ExtUtils/MM_Any.pm	2004-01-06 09:34:59.000000000 +1100
+++ perl-5.8.6/lib/ExtUtils/MM_Any.pm	2005-05-21 15:02:02.000000000 +1000
@@ -352,8 +352,6 @@
 sub manifypods_target {
     my($self) = shift;
 
-    my $man1pods      = '';
-    my $man3pods      = '';
     my $dependencies  = '';
 
     # populate manXpods & dependencies:
@@ -373,7 +371,7 @@
     foreach my $section (qw(1 3)) {
         my $pods = $self->{"MAN${section}PODS"};
         push @man_cmds, $self->split_command(<<CMD, %$pods);
-	\$(NOECHO) \$(POD2MAN) --section=$section --perm_rw=\$(PERM_RW)
+	\$(NOECHO) \$(POD2MAN) --section=\$(MAN${section}EXT) --perm_rw=\$(PERM_RW)
 CMD
     }
 
diff -Naur --exclude=debian perl-5.8.6.orig/lib/ExtUtils/MM_Unix.pm perl-5.8.6/lib/ExtUtils/MM_Unix.pm
--- perl-5.8.6.orig/lib/ExtUtils/MM_Unix.pm	2004-01-06 09:34:59.000000000 +1100
+++ perl-5.8.6/lib/ExtUtils/MM_Unix.pm	2005-05-21 15:08:26.000000000 +1000
@@ -2092,6 +2092,11 @@
         $self->{PERLPREFIX}   ||= $iprefix;
         $self->{SITEPREFIX}   ||= $sprefix;
         $self->{VENDORPREFIX} ||= $vprefix;
+	my $p = $self->{PREFIX} = $self->{PERLPREFIX};
+	for my $t (qw/PERL SITE VENDOR/)
+	{
+	    $self->{"${t}PREFIX"} =~ s!^\Q$p\E(?=/|$)!\$(PREFIX)!;
+	}
     }
 
     my $arch    = $Config{archname};
@@ -2499,9 +2504,7 @@
 	$(NOECHO) $(ECHO) INSTALLDIRS not defined, defaulting to INSTALLDIRS=site
 
 pure_perl_install ::
-	$(NOECHO) $(MOD_INSTALL) \
-		read }.$self->catfile('$(PERL_ARCHLIB)','auto','$(FULLEXT)','.packlist').q{ \
-		write }.$self->catfile('$(DESTINSTALLARCHLIB)','auto','$(FULLEXT)','.packlist').q{ \
+	$(NOECHO) umask 022; $(MOD_INSTALL) \
 		$(INST_LIB) $(DESTINSTALLPRIVLIB) \
 		$(INST_ARCHLIB) $(DESTINSTALLARCHLIB) \
 		$(INST_BIN) $(DESTINSTALLBIN) \
@@ -2513,7 +2516,7 @@
 
 
 pure_site_install ::
-	$(NOECHO) $(MOD_INSTALL) \
+	$(NOECHO) umask 02; $(MOD_INSTALL) \
 		read }.$self->catfile('$(SITEARCHEXP)','auto','$(FULLEXT)','.packlist').q{ \
 		write }.$self->catfile('$(DESTINSTALLSITEARCH)','auto','$(FULLEXT)','.packlist').q{ \
 		$(INST_LIB) $(DESTINSTALLSITELIB) \
@@ -2526,9 +2529,7 @@
 		}.$self->catdir('$(PERL_ARCHLIB)','auto','$(FULLEXT)').q{
 
 pure_vendor_install ::
-	$(NOECHO) $(MOD_INSTALL) \
-		read }.$self->catfile('$(VENDORARCHEXP)','auto','$(FULLEXT)','.packlist').q{ \
-		write }.$self->catfile('$(DESTINSTALLVENDORARCH)','auto','$(FULLEXT)','.packlist').q{ \
+	$(NOECHO) umask 022; $(MOD_INSTALL) \
 		$(INST_LIB) $(DESTINSTALLVENDORLIB) \
 		$(INST_ARCHLIB) $(DESTINSTALLVENDORARCH) \
 		$(INST_BIN) $(DESTINSTALLVENDORBIN) \
@@ -2537,37 +2538,19 @@
 		$(INST_MAN3DIR) $(DESTINSTALLVENDORMAN3DIR)
 
 doc_perl_install ::
-	$(NOECHO) $(ECHO) Appending installation info to $(DESTINSTALLARCHLIB)/perllocal.pod
-	-$(NOECHO) $(MKPATH) $(DESTINSTALLARCHLIB)
-	-$(NOECHO) $(DOC_INSTALL) \
-		"Module" "$(NAME)" \
-		"installed into" "$(INSTALLPRIVLIB)" \
-		LINKTYPE "$(LINKTYPE)" \
-		VERSION "$(VERSION)" \
-		EXE_FILES "$(EXE_FILES)" \
-		>> }.$self->catfile('$(DESTINSTALLARCHLIB)','perllocal.pod').q{
 
 doc_site_install ::
-	$(NOECHO) $(ECHO) Appending installation info to $(DESTINSTALLARCHLIB)/perllocal.pod
-	-$(NOECHO) $(MKPATH) $(DESTINSTALLARCHLIB)
-	-$(NOECHO) $(DOC_INSTALL) \
+	$(NOECHO) $(ECHO) Appending installation info to $(DESTINSTALLSITEARCH)/perllocal.pod
+	-$(NOECHO) umask 02; $(MKPATH) $(DESTINSTALLSITEARCH)
+	-$(NOECHO) umask 02; $(DOC_INSTALL) \
 		"Module" "$(NAME)" \
 		"installed into" "$(INSTALLSITELIB)" \
 		LINKTYPE "$(LINKTYPE)" \
 		VERSION "$(VERSION)" \
 		EXE_FILES "$(EXE_FILES)" \
-		>> }.$self->catfile('$(DESTINSTALLARCHLIB)','perllocal.pod').q{
+		>> }.$self->catfile('$(DESTINSTALLSITEARCH)','perllocal.pod').q{
 
 doc_vendor_install ::
-	$(NOECHO) $(ECHO) Appending installation info to $(DESTINSTALLARCHLIB)/perllocal.pod
-	-$(NOECHO) $(MKPATH) $(DESTINSTALLARCHLIB)
-	-$(NOECHO) $(DOC_INSTALL) \
-		"Module" "$(NAME)" \
-		"installed into" "$(INSTALLVENDORLIB)" \
-		LINKTYPE "$(LINKTYPE)" \
-		VERSION "$(VERSION)" \
-		EXE_FILES "$(EXE_FILES)" \
-		>> }.$self->catfile('$(DESTINSTALLARCHLIB)','perllocal.pod').q{
 
 };
 
@@ -2850,7 +2833,7 @@
 	($lperl = $libperl) =~ s/\$\(A\)/$self->{LIB_EXT}/;
     }
     unless ($libperl && -f $lperl) { # Ilya's code...
-	my $dir = $self->{PERL_SRC} || "$self->{PERL_ARCHLIB}/CORE";
+	my $dir = $self->{PERL_SRC} || "/usr/lib";
 	$dir = "$self->{PERL_ARCHLIB}/.." if $self->{UNINSTALLED_PERL};
 	$libperl ||= "libperl$self->{LIB_EXT}";
 	$libperl   = "$dir/$libperl";
diff -Naur --exclude=debian perl-5.8.6.orig/lib/ExtUtils/t/INST.t perl-5.8.6/lib/ExtUtils/t/INST.t
--- perl-5.8.6.orig/lib/ExtUtils/t/INST.t	2004-01-06 09:34:59.000000000 +1100
+++ perl-5.8.6/lib/ExtUtils/t/INST.t	2005-05-21 15:10:16.000000000 +1000
@@ -60,7 +60,7 @@
 
 my $config_prefix = $Config{installprefixexp} || $Config{installprefix} ||
                     $Config{prefixexp}        || $Config{prefix};
-is( $mm->{PERLPREFIX}, $config_prefix,   'PERLPREFIX' );
+is( $mm->{PERLPREFIX}, '$(PREFIX)',   'PERLPREFIX' );
 
 is( !!$mm->{PERL_CORE}, !!$ENV{PERL_CORE}, 'PERL_CORE' );
 
diff -Naur --exclude=debian perl-5.8.6.orig/lib/ExtUtils/t/INST_PREFIX.t perl-5.8.6/lib/ExtUtils/t/INST_PREFIX.t
--- perl-5.8.6.orig/lib/ExtUtils/t/INST_PREFIX.t	2004-01-06 09:34:59.000000000 +1100
+++ perl-5.8.6/lib/ExtUtils/t/INST_PREFIX.t	2005-05-21 15:11:23.000000000 +1000
@@ -16,7 +16,7 @@
 }
 
 use strict;
-use Test::More tests => 36;
+use Test::More tests => 32;
 use MakeMaker::Test::Utils;
 use ExtUtils::MakeMaker;
 use File::Spec;
@@ -60,9 +60,9 @@
 is( $mm->{NAME}, 'Big::Dummy',  'NAME' );
 is( $mm->{VERSION}, 0.01,            'VERSION' );
 
-foreach my $prefix (qw(PREFIX PERLPREFIX SITEPREFIX VENDORPREFIX)) {
-    unlike( $mm->{$prefix}, qr/\$\(PREFIX\)/ );
-}
+#foreach my $prefix (qw(PREFIX PERLPREFIX SITEPREFIX VENDORPREFIX)) {
+#    unlike( $mm->{$prefix}, qr/\$\(PREFIX\)/ );
+#}
 
 
 my $PREFIX = File::Spec->catdir('foo', 'bar');
